eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -372,4 +385,42 @@ protected function getDateTime($timestamp)
 
         return $dateTime;
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     *
+     * @return bool
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function userHasPermissionsToRemove(ContentInfo $contentInfo, Location $location): bool
+    {
+        if (!$this->repository->canUser('content', 'remove', $contentInfo, [$location])) {
+            return false;
+        }
+
+        $contentRemoveCriterion = $this->permissionCriterionResolver->getPermissionsCriterion('content', 'remove');
+
+        if (!$contentRemoveCriterion instanceof Criterion) {
+            return (bool)$contentRemoveCriterion;
+        }
+
+        $query = new Query(
+            array(
```

Comment:
what is the state of previous policy when in new methods / parts of code we used short array syntax?
@alongosz